### ISSUE TITLE
Internal class AccessiblePrivateMethods used in non-internal code

### ISSUE THREAD

**Issue by szepeviktor** [woothemes/woocommerce#35050]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Hello! I run a project called WC stubs. It excludes internal stuff as that should not be used by WC extension developers.

In this class an internal trait is used.
https://github.com/woocommerce/woocommerce/blob/1925cfa80eb754a4727eaf66b3336c0f77a7bb05/plugins/woocommerce/includes/class-wc-query.php#L17-L19

This brakes my stubs.
Please help me.

### Expected behavior

No internal stuff used in non-internal code.

### Actual behavior

Internal stuff is used in non-internal code.

### Steps to reproduce

1. See my CI https://app.travis-ci.com/github/php-stubs/woocommerce-stubs/jobs/585388538

### WordPress Environment

no

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by szepeviktor** [comment]
I've opened this issue for **your** benefit.
Not interested in working more than opening it.

**Comment by barryhughes** [comment]
Hi @szepeviktor,

It's not clear to us what the problem is here. 

I searched for "wc-stubs" and found two projects [[1]](https://github.com/polylang/polylang-wc-stubs) [[2]](https://github.com/polylang/wc-mix-and-match-products-stubs) but I assume they are unrelated. Though the problem must be pretty clear to you, we are really lacking context here.

Something we would note is that, "behind the scenes", our public code will indeed make use of internal code—such as in the example you shared—and that is fine (it would only be a problem if we exposed public methods or functions that require internal types as parameters, or supply them as return values—and that is not the case as far as we know).

I'm confused because in your opening post you wrote:

> This brakes my stubs.
> Please help me.

Then you followed up with:

> I've opened this issue for your benefit.
> Not interested in working more than opening it.

On that basis, I'll close this issue. If you do need help, we'll be happy to work with you: but we would need a lot more context.
